packages/open-next/src/cache/incremental/s3-lite.ts

```diff
@@ -7,24 +8,33 @@ import { parseNumberFromEnv } from "../../adapters/util";
 import { Extension } from "../next-types";
 import { IncrementalCache } from "./types";
 
-const {
-  CACHE_BUCKET_REGION,
-  CACHE_BUCKET_KEY_PREFIX,
-  NEXT_BUILD_ID,
-  CACHE_BUCKET_NAME,
-} = process.env;
+let awsClient: AwsClient | null = null;
 
-const awsClient = new AwsClient({
-  accessKeyId: process.env.AWS_ACCESS_KEY_ID!,
-  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY!,
-  sessionToken: process.env.AWS_SESSION_TOKEN,
-  region: CACHE_BUCKET_REGION,
-  retries: parseNumberFromEnv(process.env.AWS_SDK_S3_MAX_ATTEMPTS),
-});
+const getAwsClient = () => {
+  const { CACHE_BUCKET_REGION } = process.env;
+  if (awsClient) {
+    return awsClient;
+  } else {
+    awsClient = new AwsClient({
+      accessKeyId: process.env.AWS_ACCESS_KEY_ID!,
+      secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY!,
+      sessionToken: process.env.AWS_SESSION_TOKEN,
+      region: CACHE_BUCKET_REGION,
+      retries: parseNumberFromEnv(process.env.AWS_SDK_S3_MAX_ATTEMPTS),
+    });
+    return awsClient;
+  }
+};
 
-const awsFetch = customFetchClient(awsClient);
+const awsFetch = async (key: string, options: RequestInit) => {
+  const { CACHE_BUCKET_REGION, CACHE_BUCKET_NAME } = process.env;
+  const client = getAwsClient();
+  const url = `https://${CACHE_BUCKET_NAME}.s3.${CACHE_BUCKET_REGION}.amazonaws.com/${key}`;
+  return customFetchClient(client)(url, options).finally(console.log);
```

Comment:
Should we be using `debug` here instead of `console.log`?

Reply:
Yeah, it shouldn't have stayed there, let me remove this quickly